Applications/Spire/Source/KeyBindings/TaskKeysPage.cpp

```diff
@@ -73,18 +89,140 @@ TaskKeysPage::TaskKeysPage(std::shared_ptr<KeyBindingsModel> key_bindings,
   auto layout = make_vbox_layout(body);
   layout->addWidget(make_help_text_box());
   layout->addWidget(toolbar);
-  layout->addWidget(make_task_keys_table_view(
+  m_table_view = make_task_keys_table_view(
     m_key_bindings->get_order_task_arguments(),
     std::make_shared<LocalComboBoxQueryModel>(), std::move(destinations),
-    std::move(markets)));
+    std::move(markets));
+  layout->addWidget(m_table_view);
   auto box = new Box(body);
   update_style(*box, [] (auto& style) {
     style.get(Any()).set(BackgroundColor(QColor(0xFFFFFF)));
   });
   enclose(*this, *box);
+  update_button_state();
+  auto& row_selection = m_table_view->get_selection()->get_row_selection();
+  m_selection_connection = row_selection->connect_operation_signal(
+    std::bind_front(&TaskKeysPage::on_row_selection, this));
+  m_table_operation_connection =
+    m_table_view->get_body().get_table()->connect_operation_signal(
+      std::bind_front(&TaskKeysPage::on_table_operation, this));
 }
 
 const std::shared_ptr<KeyBindingsModel>&
     TaskKeysPage::get_key_bindings() const {
   return m_key_bindings;
 }
+
+bool TaskKeysPage::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == m_added_region_item) {
+    if(event->type() == QEvent::Show) {
```

Comment:
Why do you need to wait for the Show event in order to ```set_editing```?

Reply:
The `ComboBox` within the RegionBox uses a delay initialization to initialize `m_input_focus_proxy`. If the function `set_editing` was invoked at line 219, the `RegionBox` would pop up before the `ComboBox` was displayed, resulting in a crash because the `ComboBox` attempt to send events to `m_input_focus_proxy` which haven’t been initialized (Line 437 in ComboBox.cpp).

Although sending mouse events to uninitialized `m_input_focus_proxy` can be avoided by adding check code before sending events, the `RegionBox` will be displayed before the newly added row is shown. I have put a demo `KeyBindingsUiTester-RegionBox.exe` into demos/task_keys_page_operation to illustrate this case. You will see that the RegionBox is displayed briefly before the new row is shown.

Reply:
Okay, thanks.